server/main.lua

```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
-        exports.qbx_core:Notify(src, "Civilian is not in city.", "error")
+        exports.qbx_core:Notify(src, 'Civilian is not in city.', 'error')
         return
     end
 
     local success
-    if type == 'gang' then
-        success = employee.Functions.SetGang(player.PlayerData[type].name, data.grade)
+    if data.group == 'gang' then
+        success = employee.Functions.SetGang(player.PlayerData[data.group].name, data.grade)
     else
-        success = employee.Functions.SetJob(player.PlayerData[type].name, data.grade)
+        success = employee.Functions.SetJob(player.PlayerData[data.group].name, data.grade)
     end
 
     if success then
-        exports.qbx_core:Notify(src, "Successfully promoted!", "success")
-        exports.qbx_core:Notify(employee.PlayerData.source, "You have been promoted to " ..data.gradename..".", "success")
+        exports.qbx_core:Notify(src, 'Successfully promoted!', 'success')
+        exports.qbx_core:Notify(employee.PlayerData.source, 'You have been promoted to ' ..data.gradename..'.', 'success')
     else
-        exports.qbx_core:Notify(src, "Grade does not exist.", "error")
+        exports.qbx_core:Notify(src, 'Grade does not exist.', 'error')
     end
 end
 
----@param src number
+---@param closePlayers table List of nearby server ids
 ---@return table
-function GetPlayers(src)
+lib.callback.register('qbx_management:server:getplayers', function(_, closePlayers)
 	local players = {}
-	local playerPed = GetPlayerPed(src)
-	local pCoords = GetEntityCoords(playerPed)
-	for _, v in pairs(exports.qbx_core:GetPlayers()) do
-		local targetped = GetPlayerPed(v)
-		local tCoords = GetEntityCoords(targetped)
-		local dist = #(pCoords - tCoords)
-		if playerPed ~= targetped and dist < 10 then
-			local ped = exports.qbx_core:GetPlayer(tonumber(v))
-			players[#players + 1] = {
-				id = v,
-				coords = GetEntityCoords(targetped),
-				name = ped.PlayerData.charinfo.firstname .. " " .. ped.PlayerData.charinfo.lastname,
-				citizenid = ped.PlayerData.citizenid,
-				sources = GetPlayerPed(ped.PlayerData.source),
-				sourceplayer = ped.PlayerData.source
-			}
-		end
+	for _, v in pairs(closePlayers) do
+		local player = exports.qbx_core:GetPlayer(v.id)
+		players[#players + 1] = {
+			id = v.id,
+			name = player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname,
+			citizenid = player.PlayerData.citizenid,
+			job = player.PlayerData.job,
+			gang = player.PlayerData.gang,
+			source = player.PlayerData.source
+		}
 	end
 
 	table.sort(players, function(a, b)
 		return a.name < b.name
 	end)
 
 	return players
-end
+end)
+
+---@param accountName string
+---@param group 'job'|'gang'
+---@return table?
+lib.callback.register('qbx_management:server:getemployees', function(source, accountName, group)
+	return getEmployees(source, accountName, group)
+end)
+
+---@param data table
+---@return table?
+RegisterNetEvent('qbx_management:server:updateGrade', function(data)
```

Comment:
Should this be a callback returning nil instead? Events should describe something that has happened in the past https://manason.github.io/effective-fivem-lua/events/#past-tense